FireboltNETSDK/Utils/Types.cs

```diff
@@ -331,30 +354,44 @@ public static IEnumerable<NewMeta> ParseJsonResponse(string? response)
             {
                 var prettyJson = JToken.Parse(response).ToString(Formatting.Indented);
                 var data = JsonConvert.DeserializeObject<QueryResult>(prettyJson);
-                if (data == null) throw new FireboltException("Unable to parse data to JSON");
-                var newListData = new List<NewMeta>();
-                foreach (var t in data.Data)
-                    for (var j = 0; j < t.Count; j++)
-                    {
-                        var columnType = ColumnType.Of(TypesConverter.GetFullColumnTypeName(data.Meta[j]));
-                        newListData.Add(new NewMeta
-                        (
-                            data: new ArrayList
-                            {
-                                TypesConverter.ConvertToCSharpVal(t[j]?.ToString(), columnType)
-                            },
-                            meta: columnType.Type.ToString()
-                        ));
-                    }
-                return newListData;
+                return ProcessQueryResult(data);
             }
             catch (System.Exception e)
             {
                 throw new FireboltException("Error while parsing response", e);
             }
         }
+
+
+        public static IEnumerable<NewMeta> ProcessQueryResult(QueryResult? result)
+        {
+            if (result == null) throw new FireboltException("Unable to parse data to JSON");
```

Comment:
IMO, the error about parsing should be next to the parsing function, not the one that makes sense of data afterwards. This method does not know about parsing or JSON. Maybe later on we'll have tab-delimited data instead. 
I'd move this exception to ConvertToDate and if a null passed in here would change the message to something like "Can't process null values".

Reply:
I moved this validation to other implementation of this method that accepts JSON as string. Moreover method `ProcessQueryResult(QueryResult? result)` is now private and I have removed the only one reference to it that in turn was not in use.